opengrok-indexer/src/main/java/org/opengrok/indexer/web/Util.java

```diff
@@ -713,82 +713,87 @@ public static void readableLine(int num, Writer out, Annotation annotation, Stri
         out.write(closeQuotedTag);
         out.write(snum);
         out.write(anchorEnd);
+
         if (annotation != null) {
-            String r = annotation.getRevision(num);
-            boolean enabled = annotation.isEnabled(num);
-            out.write("<span class=\"blame\">");
-            if (enabled) {
-                out.write(anchorClassStart);
-                out.write("r");
-                out.write("\" style=\"background-color: ");
-                out.write(annotation.getColors().getOrDefault(r, "inherit"));
-                out.write("\" href=\"");
-                out.write(URIEncode(annotation.getFilename()));
-                out.write("?a=true&amp;r=");
-                out.write(URIEncode(r));
-                String msg = annotation.getDesc(r);
-                out.write("\" title=\"");
-                if (msg != null) {
-                    out.write(msg);
-                }
-                if (annotation.getFileVersion(r) != 0) {
-                    out.write("&lt;br/&gt;version: " + annotation.getFileVersion(r) + "/"
-                            + annotation.getRevisions().size());
-                }
-                out.write(closeQuotedTag);
-            }
-            StringBuilder buf = new StringBuilder();
-            final boolean most_recent_revision = annotation.getFileVersion(r) == annotation.getRevisions().size();
-            // print an asterisk for the most recent revision
-            if (most_recent_revision) {
-                buf.append("<span class=\"most_recent_revision\">");
-                buf.append('*');
+            writeAnnotation(num, out, annotation, userPageLink, userPageSuffix, project);
+        }
+    }
+
+    private static void writeAnnotation(int num, Writer out, Annotation annotation, String userPageLink, String userPageSuffix, String project) throws IOException {
+        String r = annotation.getRevision(num);
+        boolean enabled = annotation.isEnabled(num);
+        out.write("<span class=\"blame\">");
+        if (enabled) {
+            out.write(anchorClassStart);
+            out.write("r");
+            out.write("\" style=\"background-color: ");
+            out.write(annotation.getColors().getOrDefault(r, "inherit"));
+            out.write("\" href=\"");
+            out.write(URIEncode(annotation.getFilename()));
+            out.write("?a=true&amp;r=");
+            out.write(URIEncode(r));
+            String msg = Util.encode(annotation.getDesc(r));
```

Comment:
Given the pre-existing `if (msg != null)` check nearby below, should perhaps the new `Util.encode()` be done in that if-block?

Reply:
definitely. Should have done the testing. Thanks.